hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/AbstractScheduleHandler.java

```diff
@@ -16,75 +16,361 @@
+            @Nullable final Schedule scheduleToValidate,
+            @Nullable final Instant consensusTime,
+            final boolean isLongTermEnabled) {
+        final ResponseCodeEnum result;
+        final Instant effectiveConsensusTime = Objects.requireNonNullElse(consensusTime, Instant.MIN);
+        if (scheduleToValidate != null) {
+            if (!scheduleToValidate.executed()) {
+                if (!scheduleToValidate.deleted()) {
+                    Timestamp expiration = scheduleToValidate.calculatedExpirationTime();
+                    final Instant calculatedExpiration = ScheduleUtility.instantFromTimestamp(expiration, Instant.MAX);
+                    if (effectiveConsensusTime.isBefore(calculatedExpiration)) {
+                        result = ResponseCodeEnum.OK;
+                    } else {
+                        // We are past expiration time
+                        if (!isLongTermEnabled) {
+                            result = ResponseCodeEnum.INVALID_SCHEDULE_ID;
+                        } else {
+                            // This is not failure, it indicates the schedule should execute if it can,
+                            // or be removed if it is not executable (i.e. it lacks required signatures)
+                            result = ResponseCodeEnum.SCHEDULE_PENDING_EXPIRATION;
+                        }
+                    }
+                } else {
+                    result = ResponseCodeEnum.SCHEDULE_ALREADY_DELETED;
+                }
+            } else {
+                result = ResponseCodeEnum.SCHEDULE_ALREADY_EXECUTED;
+            }
+        } else {
+            result = ResponseCodeEnum.INVALID_SCHEDULE_ID;
+        }
+        return result;
+    }
+
+    /**
+     * Very basic transaction ID validation.
+     * This just checks that the transaction is not scheduled (you cannot schedule a schedule),
+     * that the account ID is not null (so we can fill in scheduler account),
+     * and that the start timestamp is not null (so we can fill in schedule valid start time)
+     * @param currentId a TransactionID to validate
+     * @throws PreCheckException if the transaction is scheduled, the account ID is null, or the start time is null.
+     */
+    protected void checkValidTransactionId(@Nullable final TransactionID currentId) throws PreCheckException {
+        if (currentId == null) throw new PreCheckException(ResponseCodeEnum.INVALID_TRANSACTION_ID);
+        final AccountID payer = currentId.accountID();
+        final Timestamp validStart = currentId.transactionValidStart();
+        final boolean isScheduled = currentId.scheduled();
+        if (isScheduled) throw new PreCheckException(ResponseCodeEnum.SCHEDULED_TRANSACTION_NOT_IN_WHITELIST);
+        if (payer == null) throw new PreCheckException(ResponseCodeEnum.INVALID_SCHEDULE_PAYER_ID);
+        if (validStart == null) throw new PreCheckException(ResponseCodeEnum.INVALID_TRANSACTION_START);
```

Comment:
Optional nit:
```suggestion
        if (currentId.scheduled()) throw new PreCheckException(ResponseCodeEnum.SCHEDULED_TRANSACTION_NOT_IN_WHITELIST);
        final AccountID payer = currentId.accountID();
        if (payer == null) throw new PreCheckException(ResponseCodeEnum.INVALID_SCHEDULE_PAYER_ID);
        if (currentId.transactionValidStart() == null) throw new PreCheckException(ResponseCodeEnum.INVALID_TRANSACTION_START);
```
Left `final AccountID payer = ` because it's useful to identify the current ID as the payer. The others didn't seem useful

Reply:
The assignments are there because spotless mangles the code badly without them, unfortunately.